### PR TITLE
Fix Postarchive Escaping

### PR DESCRIPTION
**Pull request by nightpool**
Previously, any post with a title that included angle brackets broke EVERYTHING.

(can you say 'yikes')

### CODE REVIEW

**Comment by hobinjk** [comment]
Given how much of a kerfuffle there was when we had `let` in a few places I don't think that template strings are a good idea unless we introduce a compilation step (which would be nice to have so that we could develop in ~~a real language~~ ES6 without worrying about our lovely and wonderful users who don't believe in updating their browsers)

**Comment by nightpool** [comment]
@hobinjk Template strings are better supported then let—the main issue we had was Safari 9 support, which template strings have.

File: Extensions/postarchive.js [hunk 1]
```diff
@@ -244,31 +265,31 @@ XKit.extensions.postarchive = {
...
-					"<div id=\"xkit-postarchive-export\" class=\"xkit-button\">Export</div>" +
-					"<div id=\"xkit-postarchive-import\" class=\"xkit-button\">Import</div>" +
-					"<div id=\"xkit-postarchive-content\" class=\"xkit-postarchive-no-post-selected\">" +
-						"Select a post from the right to get started." +
-					"</div>" +
-					"<div id=\"xkit-postarchive-remove-this\" class=\"xkit-button hidden\">remove</div>" +
-					"<div id=\"xkit-postarchive-recategorize-this\" class=\"xkit-button hidden\">recategorize</div>" +
-					"<div id=\"xkit-postarchive-rename-this\" class=\"xkit-button hidden\">rename</div>" +
-				"</div>";
+						"<div id=\"xkit-postarchive-sidebar-content\" class=\"content\">"}${
```
Comment (hobinjk) [inline Extensions/postarchive.js]: Why does this section still use concatenation in addition to the template string?

File: Extensions/postarchive.js [hunk 2]
```diff
@@ -721,25 +742,25 @@ XKit.extensions.postarchive = {
...
-								"<div class=\"post_container\">" +
-									post_contents +
-								"</div>" +
-							"</div>" +
-						"</div>" +
-					post_tags +
-				"</div>" +
-			"</div>" +
-		"</li>";
+		m_html = `${m_html}<div class="xkit-dont-run-extensions post post_full ${post_class} same_user_as_last with_permalink no_source xkit_view_on_dash_post" id="post_${data.id}"  data-post-id='${data.id}' data-root_id='${data.id}' data-tumblelog-name='${username}' data-reblog-key='${data.reblog_key}' data-type='${data.type}'>` +
```
Comment (hobinjk) [inline Extensions/postarchive.js]: Same as above, it seems harder to read than using a single template string

**Comment by hobinjk** [comment]
Functionality LGTM

File: Extensions/postarchive.js (hunk 1 above)
Comment (nightpool) [inline Extensions/postarchive.js]: because I ran the eslint autoconvert fix, lol. I'll fix it.

**Comment by nightpool** [comment]
updated to clean up the template strings autoformat

 - fix the weird "concatenate inside of a interpolation" eslint bug
 - fix the places where eslint decided to add together template strings instead of using a multi-line string. (technically this is different from a correctness standpoint, with extra newlines, which is why I think it didn't do it automatically)
 - change stuff like `postcontent = \`${post_content}foo\`` to `postcontent += 'foo'`

**Comment by hobinjk** [comment]
Tested locally, LGTM.
@homu r+

**Comment by hobinjk** [comment]
📯 
@homu r+

**Comment by hobinjk** [comment]
Oh homu.io is actually completely down :/

**Comment by nightpool** [comment]
oh no :(

**Comment by nightpool** [comment]
luckily, nearly everything we use homu for is basically built in to github now?

**Pull request merged**